packages/cardano-services/src/Program/programs/providerServer.ts

```diff
@@ -72,6 +88,18 @@ const connectionConfigs: { [k in ConnectionNames]?: Observable<PgConnectionConfi
 
 let sharedHandleProvider: HandleProvider;
 
+const selectProviderImplementation = <T extends Provider>(
+  impl: ProviderImplementation,
+  typeorm?: () => T & TypeormProvider,
+  dbsync?: () => T & Provider,
+  blockfrost?: () => T & BlockfrostProvider
```

Comment:
Nitpick/suggestion: would be nice to group in order to keep lower # of args (we might also want to add a `logger` for example) and not have to pass `undefined` when some provider option is not implemented.
```suggestion
  providerFactories: {
    typeorm?: () => T & TypeormProvider,
    dbsync?: () => T & Provider,
    blockfrost?: () => T & BlockfrostProvider
  }
```

Reply:
Nice suggestion, here 0d44798bfa387540e2db07bfab967eab043029dd